# 04. Alignment against CHM13 reference assembly (T2T Consortium)
***

### Environment setup
```bash
mkdir 5.T2TAssembly
cd 5.T2TAssembly
ln -s /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/2.ReadsTrimming/Hifiadapterfilt_PYM007/PYM007_reads.filt.fastq.gz .
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa .
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa.fai .
```
### Alignment using [pbmm2](https://github.com/PacificBiosciences/pbmm2)
```bash
module load miniconda/4.3.1
pbmm2 align --sort -j 80 --preset HIFI --log-level INFO chm13v2.0.noMT.names.fa PYM007_reads.filt.fastq.gz PYM007.T2T.pbmm2.bam
```
### Alignment coverage analysis with [mosdepth](https://github.com/brentp/mosdepth)
```bash
module load mosdepth/0.3.3 
mosdepth -t 60 -n PYM007.T2T.pbmm2 PYM007.T2T.pbmm2.bam
```
```bash
#Plot mosdepth results
wget https://github.com/brentp/mosdepth/blob/master/scripts/plot-dist.py 
module load python38/3.8.3
chmod 777 plot-dist.py
python ./plot-dist.py PYM007.T2T.pbmm2.mosdepth.global.dist.txt
```
### Contigs' consensus with [samtools](https://github.com/samtools/samtools)
```bash
module load samtools/1.16.1 
samtools consensus -f fasta -o PYM007.T2T.pbmm2.cons.fa -a PYM007.T2T.pbmm2.bam
```
### Alignment consensus statistics with [assembly-stats](https://github.com/sanger-pathogens/assembly-stats)
```bash
module load assembly-stats/1.0.1
assembly-stats PYM007.T2T.pbmm2.cons.fa > PYM007.T2T.pbmm2.cons.assemblystats
```
### T2T-CHM13 assembly reference mapping with [minimap2](https://github.com/lh3/minimap2)
```bash
module load minimap2/2.24
minimap2 -x asm5 -L --secondary=no -t 60 chm13v2.0.noMT.names.fa PYM007.T2T.pbmm2.cons.fa > PYM007.T2T.cons.mm2.paf
```
### Dot plotting of alignment using [dotPlotly](https://github.com/tpoorten/dotPlotly)
```bash
wget https://github.com/tpoorten/dotPlotly/blob/master/pafCoordsDotPlotly.R
module load r/4.0.2
chmod 777 pafCoordsDotPlotly.R
./pafCoordsDotPlotly.R -i PYM007.T2T.cons.mm2.paf -o PYM007.T2T -s -t -l -x
```
### Genome quality report with [QUAST](https://quast.sourceforge.net/)
```bash
python quast.py -t 18 --large -o T2TvsPYM -r chm13v2.0.noMT.names.fa PYM007.T2T.pbmm2.cons.fa
```
### Coverage plot with [pafr](https://cran.r-project.org/web/packages/pafr/vignettes/Introduction_to_pafr.html)
This part has to be run in R
```R
# Libraries loading
library(ggplot2)
library(pafr)
# Data uploading
CHM13 <- read_paf("PYM007.T2T.cons.mm2.paf") 
# Chromosomes ordering
CHM13$tname <- factor(CHM13$tname, levels = c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y"))
CHM13$qname <- factor(CHM13$qname, levels = c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y"))
# Plotting
mis_colores <- c('#07948e','#3d8e6f','#728751','#ECB004','#de7b13','#FE7701',
                 '#f86a29','#f36046','#ee5663','#E94B86','#ea5d6f','#ea7353',
                 '#eb8837','#eb9e1b','#e5b00d','#c0b139','#9cb265','#78b391',
                 '#54b4bd','#45B4CF','#3fa7cf','#3b9fcf','#3896cf','#348ecf')
plot_coverage(CHM13, fill='qname') + scale_fill_manual(values = mis_colores) 
```

