# 8. Comparison between the obtained alignments and assemblied genomes 
***

## 8.1. Comparison between the sample (PYM007) alignments and the human genome references 

### Bar chart of misassemblies per chromosome
To plot the bar chart it's necessary to make use of the mis_contigs.info files obtained at the [QUAST](https://quast.sourceforge.net/) quality assesment. The scripts used at this step are saved at the "scripts" directory within this repository. 
```bash
# Misassembly counting per chromosome 
python misassembliescount38.py contigs_report_PYM007-GRCh38-pbmm2-cons.mis_contigs.info
python misassembliescountT2T.py contigs_report_PYM007-T2T-pbmm2-cons.mis_contigs.info
# Bar chart plotting
python misassembliesxchr.py missasemblies_count_38.csv missasemblies_count_T2T.csv misassembliesxchrplot.png
```
### Bar chart of sequence identity per chromosome
To plot the bar chart it's necessary to make use of the icarus.html files obtained at the [QUAST](https://quast.sourceforge.net/) quality assesment. The script used at this step is saved at the "scripts" directory within this repository. 
```bash
# Bar chart plotting
python seqidentity.py genomefraq38.csv genomefraqT2T.csv genomefraqxchrplot.png
```
### Mean coverage per chromosome bar chart 
Para obtener esta grafica de barras necesitamos crear previamente un archivo .csv con la cobertura promedio por cromosoma de los alineamientos obtenidos contra las referencias GRCh38 y CHM13, que pueden encontrarse en el archivo mosdepth..summary.txt obtenido con el programa [mosdepth](https://github.com/brentp/mosdepth). Luego de eso, se debe correr el script meanchrcoverage.py que se encuentra en el "scripts" directory within this repository.
```bash
python meanchrcoverage.py mosdepthchrcoverage.csv meanchrcoverage.png
```

## 8.2. Comparison between the obtained sample (PYM007) alignments and de novo assembly. 

### Assembled vs missasembled reads
Para comparar las lecturas mapeadas de PYM007-38, PYM007-T2T y PYM007denovo se hará uso de samtools y los archivos bam (en el caso del ensamble de novo se hizo previo un mapeo de los reads vs el ensamble obtenido, reportado en la parte 3.DenovoAsembly). 
```bash
module load samtools/1.16.1
samtools flastat archivo.bam #Para cada archivo bam (38, T2T, denovo)
```

## 8.2. Comparison between the obtained alignments with GRCh38 and T2T  

### Coverage plot
Se analizó la cobertura del alineamiento obtenido con GRCh38 en el alineamiento obtenido con T2T y viceversa mediante el uso del programa [pafr](https://github.com/dwinter/pafr).
```R
library(ggplot2)
library(pafr)
library(paletteer)
paf <- read_paf("PYM007.GRCh38vsT2T.mm2.paf")
# Modificar la variable $tname por $qname para obtener el resultado contrario
paf$tname <- factor(paf$tname, levels = c("1", "2", "3", "4", "5", "6",
                                                "7", "8", "9", "10", "11", "12",
                                                "13", "14", "15", "16", "17", 
                                                "18", "19", "20", "21", "22", 
                                                "X", "Y", "-"))
# Modificar la variable $qname por $tname para obtener el resultado contrario
paf$qname <- factor(paf$qname, levels = c("1", "2", "3", "4", "5", "6", 
                                                "7", "8", "9", "10", "11", "12",
                                                "13", "14", "15", "16", "17",
                                                "18", "19", "20", "21", "22", 
                                                "X", "Y", "-"))
mis_colores <- c('#07948e','#3d8e6f','#728751','#ECB004','#de7b13','#FE7701',
                 '#f86a29','#f36046','#ee5663','#E94B86','#ea5d6f','#ea7353',
                 '#eb8837','#eb9e1b','#e5b00d','#c0b139','#9cb265','#78b391',
                 '#54b4bd','#45B4CF','#3fa7cf','#3b9fcf','#3896cf','#348ecf',
                 '#D3D3D3')
plot_coverage(paf, fill='qname') + scale_fill_manual(values = mis_colores)
```



#Preparar carpeta y archivos
```bash
mkdir 9.vsAssemblies
cd 9.vsAssemblies 
ln -s /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/3.DeNovoAssembly/PYM007_hifiasm_denovo.fa .
ln -s /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/4.GRCh38Assembly/PYM007.GRCh38.pbmm2.cons.fa .
ln -s /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.cons.fa .
```
```bash
GSAlign -r PYM007.GRCh38.pbmm2.cons.fa -q PYM007.T2T.pbmm2.cons.fa -o GS/PYM007_GRCh38vsT2T -t 20
```

#Comparar GRCh38 vs T2T
```bash
module load gsalign/29nov22
module load gnuplot/5.2.6
GSAlign -r PYM007.GRCh38.pbmm2.cons.fa -q PYM007.T2T.pbmm2.cons.fa -o PYM007_GRCh38vsT2T -t 20
```
```bash
module load minimap2/2.24
minimap2 -x asm5 -L --secondary=no -t 60 PYM007.GRCh38.pbmm2.cons.fa PYM007.T2T.pbmm2.cons.fa > PYM007.GRCh38vsT2T.mm2.paf
```
```bash
module load r/4.0.2
./pafCoordsDotPlotly.R -i PYM007.GRCh38vsT2T.mm2.paf -o PYM007.GRCh38vsT2T -s -t -l -x
```

#Comparar GRCh38 vs de novo
module load gsalign/29nov22
module load gnuplot/5.2.6
GSAlign -r PYM007.GRCh38.pbmm2.cons.fa -q PYM007_hifiasm_denovo.fa -o PYM007_GRCh38vsDenovo -t 20

module load minimap2/2.24
minimap2 -x asm5 -L --secondary=no -t 60 PYM007.GRCh38.pbmm2.cons.fa PYM007_hifiasm_denovo.fa > PYM007.GRCh38vsDenovo.mm2.paf

module load r/4.0.2
./pafCoordsDotPlotly.R -i PYM007.GRCh38vsDenovo.mm2.paf -o PYM007.GRCh38vsDenovo -s -t -l -x

#Comparar T2T vs de novo
module load gsalign/29nov22
module load gnuplot/5.2.6
GSAlign -r PYM007.T2T.pbmm2.cons.fa -q PYM007_hifiasm_denovo.fa -o PYM007_T2TvsDenovo -t 20

module load minimap2/2.24
minimap2 -x asm5 -L --secondary=no -t 60 PYM007.T2T.pbmm2.cons.fa PYM007_hifiasm_denovo.fa > PYM007.T2TvsDenovo.mm2.paf

module load r/4.0.2
./pafCoordsDotPlotly.R -i PYM007.T2TvsDenovo.mm2.paf -o PYM007.T2TvsDenovo -s -t -l -x

