# 6. Single Nucleotide Variants Calling of GRCh38 Alignment
***

### Environment setup
```bash
mkdir 6.SNVcalling38
cd 6.SNVcalling38
#Deepvariant has some problems with dynamic links, so we have to copy the files directly to the folder 
cp /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.names.fasta . 
cp /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.names.fasta.fai .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/4.GRCh38Assembly/PYM007.GRCh38.pbmm2.bam .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/4.GRCh38Assembly/PYM007.GRCh38.pbmm2.bam.bai . 
```
### Variant calling using [DeepVariant](https://github.com/google/deepvariant)
```bash
#It's necessary to previously configure a docker or a closed environment, for more information check out DeepVariant documentation 
module load singularity/3.7.0
singularity run \
-B /cm/shared/apps/singularity/images/3.7.0/deepvariant_1.4.0.sif \
-B /usr/lib/locale/:/usr/lib/locale/ \
-B /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38 \
docker://google/deepvariant:1.4.0 \
/opt/deepvariant/bin/run_deepvariant \
--model_type PACBIO \
--ref /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38/Homo_sapiens_GRCh38.p14.noMT.names.fasta \
--reads /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38/PYM007.GRCh38.pbmm2.bam \
--output_vcf /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38/PYM007_GRCh38_deepvariant.vcf \
--num_shards 40
```

### Estadisticas bÃ¡sicas de VCF con [bcftools](https://samtools.github.io/bcftools/bcftools.html)
```bash
module load bcftools/1.10.2
module load python38/3.8.3
bcftools stats PYM007_GRCh38_deepvariant.vcf > PYM007_GRCh38_bcftools.vchk
plot-vcfstats -p bcftools -t "PYM007vs38_bcftools" PYM007_GRCh38_bcftools.vchk
```

```bash
module load vcftools/0.1.14
```
