# 7. Single Nucleotide Variants Calling of CHM13 Alignment (T2T consortium)
***

### Environment setup
```bash
mkdir 7.SNVcallingT2T
cd 7.SNVcallingT2T
#Deepvariant has some problems with dynamic links, so we have to copy the files directly to the folder 
cp /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa  . 
cp /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa.fai .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam.bai . 
```
### Variant calling using [DeepVariant](https://github.com/google/deepvariant)
```bash
#It's necessary to previously configure a docker or a closed environment, for more information check out DeepVariant documentation 
module load singularity/3.7.0
singularity run \
-B /cm/shared/apps/singularity/images/3.7.0/deepvariant_1.4.0.sif \
-B /usr/lib/locale/:/usr/lib/locale/ \
-B /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T \
docker://google/deepvariant:1.4.0 \
/opt/deepvariant/bin/run_deepvariant \
--model_type PACBIO \
--ref /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/chm13v2.0.noMT.names.fa \
--reads /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/PYM007.T2T.pbmm2.bam \
--output_vcf /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/PYM007_T2T_deepvariant.vcf \
--num_shards 40
```

### Estadisticas básicas de VCF con [bcftools](https://samtools.github.io/bcftools/bcftools.html)
```bash
module load bcftools/1.10.2
module load python38/3.8.3
bcftools stats PYM007_T2T_deepvariant.vcf > PYM007_T2T_bcftools.vchk
plot-vcfstats -p bcftools -t "PYM007vsT2T_bcftools" PYM007_T2T_bcftools.vchk

#Calcular depth metrics
bcftools query -f '[%DP]\n' PYM007_T2T_deepvariant.noRC.vcf > depth.txt
awk 'NR==1 || $1 < min {min=$1} END {print min}' depth.txt #profundidad mínima
awk 'NR==1 || $1 > max {max=$1} END {print max}' depth.txt #profundidad máxima
awk '{sum+=$1} END {print sum/NR}' depth.txt #profundidad promedio

#Calcular qual metrics
bcftools query -f '%QUAL\n' PYM007_T2T_deepvariant.noRC.vcf > qual.txt
awk 'NR==1 || $1 < min {min=$1} END {print min}' qual.txt #QUAL min
awk 'NR==1 || $1 > max {max=$1} END {print max}' qual.txt #QUAL max
awk '{sum+=$1} END {print sum/NR}' qual.txt #QUAL promedio

#Calcular GQ metrics
bcftools query -f '[%GQ]\n' PYM007_T2T_deepvariant.noRC.vcf > gq.txt
awk 'NR==1 || $1 < min {min=$1} END {print min}' gq.txt #GQ mínima
awk 'NR==1 || $1 > max {max=$1} END {print max}' gq.txt #GQ máxima
awk '{sum+=$1} END {print sum/NR}' gq.txt #GQ promedio
```

### Estadisticas básicas de VCF con [GATK VariantEval](https://gatk.broadinstitute.org/hc/en-us/articles/360040507171-VariantEval-BETA)
```bash
# Hacer copias de genoma de referencia
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa .
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa.fai .
# Crear archivo .dict 
module load picard/2.6.0
picard CreateSequenceDictionary R=chm13v2.0.noMT.names.fa O=chm13v2.0.noMT.names.dict
# Correr GATK
module load gatk/4.2.3.0
gatk VariantEval -R chm13v2.0.noMT.names.fa --eval PYM007_T2T_deepvariant.vcf -O PYM007_T2T_gatk.txt
```

### Estadisticas básicas de VCF con [VCFstats](https://pwwang.github.io/vcfstats/)
```bash
# Instalar VCFstats
pip install -U vcfstats

# Tipo de variantes por cromosoma
vcfstats --vcf PYM007_T2T_deepvariant.noRC.vcf \
    --formula 'COUNT(1, group=VARTYPE) ~ CHROM' \
    --outdir . \
    --title 'Tipo de variantes por cromosoma' \
    --config config.toml \
    --ggs 'scale_x_discrete(name ="Cromosoma", \
        limits=["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X","Y"]); \
        ylab("No. de variantes")'

# Conteo de genotipos mutantes
vcfstats --vcf PYM007_T2T_deepvariant.noRC.vcf \
    --outdir . \
    --formula 'COUNT(1, group=GTTYPEs[HET,HOM_ALT]{0}) ~ CHROM' \
    --title 'Genotipos mutantes por cromosoma' \
    --config config.toml \
    --ggs 'scale_x_discrete(name ="Cromosoma", \
        limits=["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X","Y"]); \
        ylab("No. de variantes")'

# GQ vs depth
vcfstats --vcf PYM007_T2T_deepvariant.noRC.vcf \
    --outdir . \
    --formula 'MEAN(GQs{0}) ~ MEAN(DEPTHs{0}, group=CHROM)' \
    --title 'Calidad de genotipo vs profundidad' \
    --config config.toml \
    --ggs 'xlab("Profundidad Promedio"); \
        ylab("Calidad del Genotipo Promedio"); \
        scale_color_manual(values=["#1E90FF", "#E31A1C", "#228B22", "#6A3D9A", "#FF7F00", "#000000", "#90682F", "#6CA6CD", "#FB9A99", "#98FB98", "#CAB2D6", "#FDBF6F", "#B3B3B3", "#C3B091", "#800000", "#DA70D6", "#FF1493", "#0000FF", "#4682B4", "#00CED1", "#00FF00", "#CCCC00", "#FFD700", "#8B4500", "#A52A2A"], \
        labels=["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y"])'

# Overall distribution of allele frequency
vcfstats --vcf PYM007_T2T_deepvariant.noRC.vcf \
    --outdir . \
    --formula 'AAF ~ 1' \
    --title 'Distribución general de frecuencia alélica' \
    --config config.toml
```

### Density plot de SNVs con [karyoploteR](https://bernatgel.github.io/karyoploter_tutorial/) 
Nota: El archivo bed de citobandas de T2T se obtuvo en el [github oficial de CHM13](https://github.com/marbl/CHM13/issues/47), mientras que el ideograma se instaló siguiendo las instrucciones de [bioconductor](https://bioconductor.org/packages/release/data/annotation/html/BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0.html)
```r
# Instalar paquetes 
if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
    BiocManager::install("karyoploteR", force = TRUE)
    BiocManager::install("VariantAnnotation", force = TRUE)
    BiocManager::install("GenomicRanges")
    BiocManager::install("BSgenome", force = TRUE) options(timeout = 600)  # Aumenta el tiempo de espera a 600 segundos
    BiocManager::install("BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0", force = TRUE) #Cargar ideograma de T2T
    BiocManager::install("rtracklayer", force = TRUE)

#Cargar librerias
library(karyoploteR)
library(VariantAnnotation)
library(GenomicRanges)
library(BSgenome)
library(BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0)
library(rtracklayer)

# Cargar bed de citobandas 
cytobands_file <- "chm13v2.0_cytobands_allchrs.bed" #descargado de github previamente

# Cargar el archivo de citobandas usando import
cytobands_data <- read.table(cytobands_file, header = FALSE, stringsAsFactors = FALSE, sep = "\t",
                             col.names = c("chrom", "start", "end", "name", "gieStain"))
#Cargar archivo vcf
vcf_file <- "PYM007_T2T_deepvariant.noRC.vcf"
vcf <- readVcf(vcf_file, "BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0")

# Convertir el VCF a GRanges
gr <- rowRanges(vcf)

# Crear el karyotype plot con el genoma T2T y agregar las citobandas
kp <- plotKaryotype(genome = "BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0", cytobands = cytobands_data)
kp <- kpPlotDensity(kp, gr)
```
