# 6. Single Nucleotide Variants Calling of CHM13 Alignment (T2T consortium)
***

### Environment setup
```bash
mkdir 7.SNVcallingT2T
cd 7.SNVcallingT2T
#Deepvariant has some problems with dynamic links, so we have to copy the files directly to the folder 
cp /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa  . 
cp /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa.fai .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam.bai . 
```
### Variant calling using [DeepVariant](https://github.com/google/deepvariant)
```bash
#It's necessary to previously configure a docker or a closed environment, for more information check out DeepVariant documentation 
module load singularity/3.7.0
singularity run \
-B /cm/shared/apps/singularity/images/3.7.0/deepvariant_1.4.0.sif \
-B /usr/lib/locale/:/usr/lib/locale/ \
-B /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T \
docker://google/deepvariant:1.4.0 \
/opt/deepvariant/bin/run_deepvariant \
--model_type PACBIO \
--ref /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/chm13v2.0.noMT.names.fa \
--reads /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/PYM007.T2T.pbmm2.bam \
--output_vcf /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/PYM007_T2T_deepvariant.vcf.gz \
--num_shards 40
```

#Estadísticas básicas de VCF 
module load bcftools/1.10.2
module load python38/3.8.3
bcftools stats PYM007_T2T_deepvariant.vcf.gz > PYM007_T2T_vcfstats.vchk
plot-vcfstats -p vcfstatsT2T PYM007_T2T_vcfstats.vchk
