# 7. Single Nucleotide Variants Calling of CHM13 Alignment (T2T consortium)
***

### Environment setup
```bash
mkdir 7.SNVcallingT2T
cd 7.SNVcallingT2T
#Deepvariant has some problems with dynamic links, so we have to copy the files directly to the folder 
cp /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa  . 
cp /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa.fai .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam.bai . 
```
### Variant calling using [DeepVariant](https://github.com/google/deepvariant)
```bash
#It's necessary to previously configure a docker or a closed environment, for more information check out DeepVariant documentation 
module load singularity/3.7.0
singularity run \
-B /cm/shared/apps/singularity/images/3.7.0/deepvariant_1.4.0.sif \
-B /usr/lib/locale/:/usr/lib/locale/ \
-B /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T \
docker://google/deepvariant:1.4.0 \
/opt/deepvariant/bin/run_deepvariant \
--model_type PACBIO \
--ref /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/chm13v2.0.noMT.names.fa \
--reads /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/PYM007.T2T.pbmm2.bam \
--output_vcf /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/7.SNVcallingT2T/PYM007_T2T_deepvariant.vcf \
--num_shards 40
```

### Estadisticas básicas de VCF con [bcftools](https://samtools.github.io/bcftools/bcftools.html)
```bash
module load bcftools/1.10.2
module load python38/3.8.3
bcftools stats PYM007_T2T_deepvariant.vcf > PYM007_T2T_bcftools.vchk
plot-vcfstats -p bcftools -t "PYM007vsT2T_bcftools" PYM007_T2T_bcftools.vchk
```

### Estadisticas básicas de VCF con [GATK VariantEval](https://gatk.broadinstitute.org/hc/en-us/articles/360040507171-VariantEval-BETA)
```bash
# Hacer copias de genoma de referencia
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa .
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.names.fa.fai .
# Crear archivo .dict 
module load picard/2.6.0
picard CreateSequenceDictionary R=chm13v2.0.noMT.names.fa O=chm13v2.0.noMT.names.dict
# Correr GATK
module load gatk/4.2.3.0
gatk VariantEval -R chm13v2.0.noMT.names.fa --eval PYM007_T2T_deepvariant.vcf -O PYM007_T2T_gatk.txt
```
