# 7. Single Nucleotide Variants Calling of GRCh38 Alignment
***

### Environment setup
```bash
mkdir 6.SNVcalling38
cd 6.SNVcalling38
#Deepvariant has some problems with dynamic links, so we have to copy the files directly to the folder 
cp /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.names.fasta . 
cp /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.names.fasta.fai .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/4.GRCh38Assembly/PYM007.GRCh38.pbmm2.bam .
cp /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/4.GRCh38Assembly/PYM007.GRCh38.pbmm2.bam.bai . 
```
### Variant calling using [DeepVariant](https://github.com/google/deepvariant)
```bash
#It's necessary to previously configure a docker or a closed environment, for more information check out DeepVariant documentation 
module load singularity/3.7.0
singularity run \
-B /cm/shared/apps/singularity/images/3.7.0/deepvariant_1.4.0.sif \
-B /usr/lib/locale/:/usr/lib/locale/ \
-B /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38 \
docker://google/deepvariant:1.4.0 \
/opt/deepvariant/bin/run_deepvariant \
--model_type PACBIO \
--ref /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38/Homo_sapiens_GRCh38.p14.noMT.names.fasta \
--reads /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38/PYM007.GRCh38.pbmm2.bam \
--output_vcf /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/6.SNVcalling38/PYM007_GRCh38_deepvariant.vcf \
--num_shards 40
```

### Estadisticas básicas de VCF con [bcftools](https://samtools.github.io/bcftools/bcftools.html)
```bash
module load bcftools/1.10.2
module load python38/3.8.3

bcftools merge 0001.vcf PYM007_T2T_deepvariant.lift.unmapped.noRC.vcf -o CHM13only.vcf

bcftools stats PYM007_GRCh38_deepvariant.noRC.vcf > PYM007_GRCh38_bcftools.vchk
plot-vcfstats -p bcftools -t "PYM007vs38_bcftools" PYM007_GRCh38_bcftools.vchk

#Calcular depth metrics
bcftools query -f '[%DP]\n' PYM007_GRCh38_deepvariant.noRC.vcf > depth.txt
bcftools query -f '[%DP]\n' 0002.vcf > depth_shared.txt

awk 'NR==1 || $1 < min {min=$1} END {print min}' depth.txt #profundidad mínima
awk 'NR==1 || $1 < min {min=$1} END {print min}' depth_shared.txt #profundidad mínima

awk 'NR==1 || $1 > max {max=$1} END {print max}' depth.txt #profundidad máxima
awk 'NR==1 || $1 > max {max=$1} END {print max}' depth_shared.txt #profundidad máxima

awk '{sum+=$1} END {print sum/NR}' depth.txt #profundidad promedio
awk '{sum+=$1} END {print sum/NR}' depth_shared.txt #profundidad promedio

#Calcular qual metrics
bcftools query -f '%QUAL\n' PYM007_GRCh38_deepvariant.noRC.vcf > qual.txt
awk 'NR==1 || $1 < min {min=$1} END {print min}' qual.txt #QUAL min
awk 'NR==1 || $1 > max {max=$1} END {print max}' qual.txt #QUAL max
awk '{sum+=$1} END {print sum/NR}' qual.txt #QUAL promedio

#Calcular GQ metrics
bcftools query -f '[%GQ]\n' PYM007_GRCh38_deepvariant.noRC.vcf > gq.txt
awk 'NR==1 || $1 < min {min=$1} END {print min}' gq.txt #GQ mínima
awk 'NR==1 || $1 > max {max=$1} END {print max}' gq.txt #GQ máxima
awk '{sum+=$1} END {print sum/NR}' gq.txt #GQ promedio
```

### Estadisticas básicas de VCF con [GATK VariantEval](https://gatk.broadinstitute.org/hc/en-us/articles/360040507171-VariantEval-BETA)
```bash
# Hacer copias de genoma de referencia
ln -s /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.names.fasta .
ln -s /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.names.fasta.fai .
# Crear archivo .dict 
module load picard/2.6.0
picard CreateSequenceDictionary R=Homo_sapiens_GRCh38.p14.noMT.names.fasta O=Homo_sapiens_GRCh38.p14.noMT.names.dict
# Correr GATK
module load gatk/4.2.3.0
gatk VariantEval -R Homo_sapiens_GRCh38.p14.noMT.names.fasta --eval PYM007_GRCh38_deepvariant.noRC.vcf -O PYM007_GRCh38_gatk.txt
```

### Estadisticas básicas de VCF con [vcfstats](https://pwwang.github.io/vcfstats/)
```bash
# Instalar VCFstats
pip install -U vcfstats

# Tipo de variantes por cromosoma
vcfstats --vcf PYM007_GRCh38_deepvariant.noRC.vcf \
    --formula 'COUNT(1, group=VARTYPE) ~ CHROM' \
    --outdir . \
    --title 'Tipo de variantes por cromosoma' \
    --config config.toml \
    --ggs 'scale_x_discrete(name ="Cromosoma", \
        limits=["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X","Y"]); \
        ylab("No. de variantes")'

# Conteo de genotipos mutantes
vcfstats --vcf PYM007_GRCh38_deepvariant.noRC.vcf \
    --outdir . \
    --formula 'COUNT(1, group=GTTYPEs[HET,HOM_ALT]{0}) ~ CHROM' \
    --title 'Genotipos mutantes por cromosoma' \
    --config config.toml \
    --ggs 'scale_x_discrete(name ="Cromosoma", \
        limits=["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","X","Y"]); \
        ylab("No. de variantes")'

# GQ vs depth
vcfstats --vcf PYM007_GRCh38_deepvariant.noRC.vcf \
    --outdir . \
    --formula 'MEAN(GQs{0}) ~ MEAN(DEPTHs{0}, group=CHROM)' \
    --title 'Calidad de genotipo vs profundidad' \
    --config config.toml \
    --ggs 'xlab("Profundidad Promedio"); \
        ylab("Calidad del Genotipo Promedio"); \
        scale_color_manual(values=["#1E90FF", "#E31A1C", "#228B22", "#6A3D9A", "#FF7F00", "#000000", "#90682F", "#6CA6CD", "#FB9A99", "#98FB98", "#CAB2D6", "#FDBF6F", "#B3B3B3", "#C3B091", "#800000", "#DA70D6", "#FF1493", "#0000FF", "#4682B4", "#00CED1", "#00FF00", "#CCCC00", "#FFD700", "#8B4500", "#A52A2A"], \
        labels=["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y"])'
```

### Density plot de SNVs con [karyoploteR](https://bernatgel.github.io/karyoploter_tutorial/)
```r
# Instalar paquetes 
if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
    BiocManager::install("karyoploteR", force = TRUE)
    BiocManager::install("VariantAnnotation", force = TRUE)
    BiocManager::install("GenomicRanges")
#Cargar librerias
library(karyoploteR)
library(VariantAnnotation)
library(GenomicRanges)

#Subir vcf 
vcf_file38 <- "PYM007_GRCh38_deepvariant.noRC.vcf"
vcf38 <- readVcf(vcf_file38, "hg38")

#Convertir vcf a formato valido (GRanges)
gr38 <- rowRanges(vcf38)

#Cambiar nombre de cromosomas (agregar "chr")
seqlevels(gr38) <- paste0("chr", seqlevels(gr38))

#Graficar el cariotipo
kp <- plotKaryotype(genome="hg38")

#Agregar la densidad de variantes al cariotipo
kp <- kpPlotDensity(kp, gr38)
```






