# Liftover de ensamble contra CHM13-T2T a coordenadas de GRCh38

## Liftover con leviosam2

Crear carpeta, mover genomas, indices y bam
```
mkdir 8.Liftover
cd 8.Liftover
ln -s /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.fasta .
ln -s /mnt/Timina/cgonzaga/resources/GRCh38.14/Homo_sapiens_GRCh38.p14.noMT.fasta.fai .
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.fa .
ln -s /mnt/Timina/cgonzaga/resources/T2T_2.0/chm13v2.0.noMT.fa.fai .
ln -s /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam .
ln -s /mnt/Timina/cgonzaga/adiaz/PacBio_secuencias/PYM007/5.T2TAssembly/PYM007.T2T.pbmm2.bam.bai .
```

Descargar archivo chain 
```
wget https://hgdownload.soe.ucsc.edu/goldenPath/hs1/liftOver/chm13v2-hg38.over.chain.gz
```

Cambiar encabezados de GRCh38 para que coincidan con archivo chain 
```
sed -i 's/CM000663.2/chr1/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000664.2/chr2/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000665.2/chr3/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000666.2/chr4/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000667.2/chr5/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000668.2/chr6/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000669.2/chr7/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000670.2/chr8/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000671.2/chr9/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000672.2/chr10/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000673.2/chr11/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000674.2/chr12/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000675.2/chr13/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000676.2/chr14/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000677.2/chr15/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000678.2/chr16/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000679.2/chr17/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000680.2/chr18/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000681.2/chr19/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000682.2/chr20/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000683.2/chr21/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000684.2/chr22/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000685.2/chrX/g' Homo_sapiens_GRCh38.p14.noMT.fasta
sed -i 's/CM000686.2/chrY/g' Homo_sapiens_GRCh38.p14.noMT.fasta
```

## Liftover con UCSC liftOver
```
module load 
liftOver PYM007.T2T.pbmm2.bam chm13v2-hg38.over.chain.gz PYM007_T2TtoGRCh38_lifted.bam unmapped
```

## Liftover con CrossMap

```
wget https://hgdownload.gi.ucsc.edu/hubs/GCA/009/914/755/GCA_009914755.4/liftOver/chm13v2-hg38.over.chain.gz
module load anaconda3/2021.05
source activate crossmap64
CrossMap.py vcf chm13v2-hg38.over.chain.gz PYM004_T2T_deeptrio.vcf.gz Homo_sapiens_GRCh38.p14.noMT.name1.fasta PYM004_liftover.vcf.gz
```

## AnÃ¡lisis de coincidencias y diferencias entre alineamientos con bcftools 
```bash
# Comprimir archivos
bgzip PYM007_T2T_deepvariant.lift.vcf > PYM007_T2T_deepvariant.lift.vcf.gz
# Crear indices para vcfs
module load bcftools/1.10.2
bcftools index PYM007_T2T_deepvariant.lift.vcf.gz > PYM007_T2T_deepvariant.lift.csi
# Identificar intersecciones
bcftools isec isec.a.vcf.gz isec.b.vcf.gz -p dir
```




